test/client/integration.test.ts

```diff
@@ -45,24 +54,88 @@ describe("Client Integration", function() {
         await storagePacker.state.commit();
-        await storageSyncer.state.commit();
 
         const parameters = PRODUCTION_PARAMS;
         parameters.USE_BURN_AUCTION = false;
-        parameters.GENESIS_STATE_ROOT = storagePacker.state.root;
+        // TODO Check that this change is correct and makes sense.
+        parameters.GENESIS_STATE_ROOT = storageSyncer.state.root;
 
         const contracts = await deployAll(signer, parameters);
+
+        // Setup L1
         for (const user of group.userIterator()) {
-            // Setup L1 pubkeys
+            // Pubkey
             await contracts.blsAccountRegistry.register(user.pubkey);
+            // Approve token transfer
+            await contracts.exampleToken.approve(
+                contracts.depositManager.address,
+                initialBalance.l1Value
+            );
+            // Deposit
+            await contracts.depositManager.depositFor(
+                user.pubkeyID,
+                initialBalance.l1Value,
+                tokenID
+            );
         }
+        // Submit deposits
+        const stateTree = StateTree.new(parameters.MAX_DEPTH);
+        group.connect(stateTree);
+        const subtreeSize = 1 << parameters.MAX_DEPOSIT_SUBTREE_DEPTH;
+        const genCommit = new GenesisCommitment(parameters.GENESIS_STATE_ROOT);
+        const genesisBatch = new ConcreteBatch([genCommit]);
+        const genesisProof = genesisBatch.proofCompressed(0);
+        await Array.from(group.groupInterator(subtreeSize)).reduce(
+            async (prevProofPromise, curSubGroup, i) => {
+                const prevProof = await prevProofPromise;
+
+                const mergeOffsetLower = i * subtreeSize;
+                const vacant = stateTree.getVacancyProof(
+                    mergeOffsetLower,
+                    parameters.MAX_DEPOSIT_SUBTREE_DEPTH
+                );
+
+                await contracts.rollup.submitDeposits(prevProof, vacant, {
+                    value: parameters.STAKE_AMOUNT
+                });
+
+                curSubGroup.createStates({
+                    initialBalance: initialBalance.l2Value,
+                    tokenID,
+                    zeroNonce: true
+                });
+                // This is required by DepositCommitment but is not used in generating proof
+                const context = {
+                    subtreeID: -1,
+                    depositSubtreeRoot: "fake",
+                    pathToSubTree: -1
+                };
+                console.warn("11111111111 new deposit root", i, stateTree.root);
+                const commitment = new DepositCommitment(
+                    stateTree.root,
+                    context
+                );
+                const depositBatch = new ConcreteBatch([commitment]);
+                const depositProof = depositBatch.proofCompressed(0);
+
+                return depositProof;
+            },
+            Promise.resolve(genesisProof)
+        );
```

Comment:
Most if not all of this can be replaced once we have the deposit packer/proposer functionality in place.